### DISCUSSION TITLE
How to get the corresponding data structure of word document

### DISCUSSION THREAD

**Discussion by yanglin2000168** [ONLYOFFICE/DocumentServer#1335] [Q&A]
How to get the corresponding data structure of word document through document server

**Comment by ShockwaveNN** [comment]
Could you explain in more details what do you want? 
We use internal format to describe document and this format is not for user usage, only to internal mechanics, so it's better to use docx format as representation of document structure

**Comment by yanglin2000168** [comment]
using java to replace the dynamic variables defined in the template







At 2021-07-08 15:08:42, "Pavel Lobashov" ***@***.***> wrote:

Could you explain in more details what do you want?
We use internal format to describe document and this format is not for user usage, only to internal mechanics, so it's better to use docx format as representation of document structure

**Comment by ShockwaveNN** [comment]
I still do not understand that you want, you're using very general syntax for seems rather complicated and very special task

**Comment by yanglin2000168** [comment]
example:
https://api.onlyoffice.com/plugin/example/invoices
${InvoiceNumber} replace  actual value，by javascript or java
















At 2021-07-08 15:17:28, "Pavel Lobashov" ***@***.***> wrote:

I still do not understand that you want, you're using very general syntax for seems rather complicated and very special task

**Reply by ShockwaveNN** [reply]
I still do not understand you

Please describe it in detailed sentence

I think this is cannot be done via Java since we use Javascript in our plugins

And via Javascript - you just call javascript methods, I do not sure that is the problem

**Comment by ShockwaveNN** [comment]
Moved this issue to discussion, since this is not an issue but a question
